sqs_notifications.tf

```diff
@@ -1,9 +1,12 @@
 locals {
-  sqs_notifications_enabled = module.this.enabled && var.bucket_notifications_enabled && var.bucket_notifications_type == "SQS"
+  enabled                   = module.this.enabled
+  sqs_notifications_enabled = local.enabled && var.bucket_notifications_enabled && var.bucket_notifications_type == "SQS"
   sqs_queue_name            = module.this.id
+  partition                 = join("", data.aws_partition.current.*.partition)
 }
 
-data "aws_caller_identity" "current" {}
+data "aws_caller_identity" "current" { count = local.enabled ? 1 : 0 }
```

Comment:
```suggestion
data "aws_caller_identity" "current" {
  count = local.enabled ? 1 : 0
}
```

Reply:
I like 1-liners for trivial stuff like this, and `terraform fmt` approves.